plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx

```diff
@@ -121,8 +137,179 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
 		return <Spinner />;
 	}
 
+	const templatePartEditUri = getSetting(
+		'templatePartProductFiltersOverlayEditUri',
+		''
+	) as string;
+
 	return (
 		<div { ...blockProps }>
+			<InspectorControls>
+				<PanelBody title={ __( 'Overlay', 'woocommerce' ) }>
+					<ToggleGroupControl
+						className="wc-block-editor-product-filters__overlay-toggle"
+						isBlock={ true }
+						value={ attributes.overlay }
+						onChange={ ( value: 'never' | 'mobile' | 'always' ) => {
+							setAttributes( { overlay: value } );
+						} }
+					>
+						<ToggleGroupControlOption
+							value={ 'never' }
+							label={ __( 'Never', 'woocommerce' ) }
+						/>
+						<ToggleGroupControlOption
+							value={ 'mobile' }
+							label={ __( 'Mobile', 'woocommerce' ) }
+						/>
+						<ToggleGroupControlOption
+							value={ 'always' }
+							label={ __( 'Always', 'woocommerce' ) }
+						/>
+					</ToggleGroupControl>
+					{ ( attributes.overlay === 'always' ||
+						attributes.overlay === 'mobile' ) && (
+						<p className="wc-block-editor-product-filters__overlay-link">
+							<ExternalLink href={ templatePartEditUri }>
+								{ __( 'Edit overlay', 'woocommerce' ) }
+							</ExternalLink>
+						</p>
+					) }
+					{ ( attributes.overlay === 'mobile' ||
+						attributes.overlay === 'always' ) && (
```

Comment:
Just a preference and completely optional, but this can reduce to single condition. But using not condition is not as explicit and straight forward as yours.

```suggestion
					{ ( attributes.overlay !== 'never' ) && (
```